go/adbc/driver/bigquery/record_reader.go

```
@@ -366,4 +366,160 @@ func (e emptyArrowIterator) SerializedArrowSchema() []byte {
+	fields := make([]arrow.Field, len(l.schema))
+	for i, field := range l.schema {
+		f, err := buildField(field, 0)
+		if err != nil {
+			log.Fatalf("Error building field %s: %v", field.Name, err)
+		}
+		fields[i] = f
+	}
+
+	arrowSchema := arrow.NewSchema(fields, nil)
+
+	var buf bytes.Buffer
+	_ = ipc.NewWriter(&buf, ipc.WithSchema(arrowSchema))
+
+	return buf.Bytes()
+}
+
+func rowsToArrowRecordBatch(schema bigquery.Schema, rows [][]bigquery.Value, alloc memory.Allocator) (arrow.Record, error) {
+	if len(rows) == 0 {
+		return nil, fmt.Errorf("no rows to convert")
+	}
+
+	// Build a schema
+	fields := make([]arrow.Field, len(schema))
+	for i, field := range schema {
+		f, err := buildField(field, 0)
+		if err != nil {
+			return nil, err
+		}
+		fields[i] = f
+	}
+	arrowSchema := arrow.NewSchema(fields, nil)
+
+	// Build arrays for columns
+	builders := make([]array.Builder, len(schema))
+	for i, field := range fields {
+		builders[i] = array.NewBuilder(alloc, field.Type)
+	}
+	defer func() {
+		for _, b := range builders {
+			b.Release()
+		}
+	}()
+
+	// Populate data
+	for _, row := range rows {
+		for colIdx, val := range row {
+			if val == nil {
+				builders[colIdx].AppendNull()
+				continue
+			}
+
+			switch builder := builders[colIdx].(type) {
+			case *array.Date32Builder:
+				// BigQuery returns civil.Date for DATE columns
+				if d, ok := val.(civil.Date); ok {
+					t := time.Date(d.Year, time.Month(d.Month), d.Day, 0, 0, 0, 0, time.UTC)
+					builder.Append(arrow.Date32FromTime(t))
+				} else if t, ok := val.(time.Time); ok {
+					builder.Append(arrow.Date32FromTime(t))
+				} else {
+					builder.AppendNull()
+				}
+			case *array.TimestampBuilder:
+				if ts, ok := val.(time.Time); ok {
+					builder.Append(arrow.Timestamp(ts.UnixMicro()))
+				} else {
+					builder.AppendNull()
+				}
+			// TODO: Add support for other types as needed
+			default:
+				if sb, ok := builder.(*array.StringBuilder); ok {
```

Can we return an error here instead? Otherwise we will get mysterious issues on the Rust side.

It's really sad that we have to do this for BigQuery as well :/